File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const hash = {};

    for (const str of strs) {
        const s = str.split('').sort().join('');
        if (!hash[s]) hash[s] = [];
        hash[s].push(str);
    }

    return Object.values(hash);
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagramDict = {}
        
        for str in strs:
            anagramKey = ''.join(sorted(str))

            if anagramKey in anagramDict:
                anagramDict[anagramKey].append(str)
            else:
                anagramDict[anagramKey] = [str]
                
        return list(anagramDict.values())